javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java

```diff
@@ -795,34 +793,99 @@ void apply(NodeText nodeText, Node node) {
         } while (diffIndex < this.elements.size() || nodeTextIndex < nodeText.getElements().size());
     }
 
-    private int findIndexOfCorrespondingNodeTextElement(CsmElement csmElement, NodeText nodeText, int startIndex, Set<Integer> usedIndexes, Node node) {
-        for (int i=startIndex;i<nodeText.getElements().size();i++){
-            if (!usedIndexes.contains(i)) {
-                TextElement textElement = nodeText.getTextElement(i);
-                if (csmElement instanceof CsmToken) {
-                    CsmToken csmToken = (CsmToken)csmElement;
-                    if (textElement instanceof TokenTextElement) {
-                        TokenTextElement tokenTextElement = (TokenTextElement)textElement;
-                        if (tokenTextElement.getTokenKind() == csmToken.getTokenType() && tokenTextElement.getText().equals(csmToken.getContent(node))) {
-                            usedIndexes.add(i);
-                            return i;
+    private List<Integer> findIndexOfCorrespondingNodeTextElement(List<CsmElement> elements, NodeText nodeText, int startIndex, Node node) {
+        List<Integer> correspondingIndices = new ArrayList<>();
+        for (ListIterator<CsmElement> csmElementListIterator = elements.listIterator(); csmElementListIterator.hasNext(); ) {
+
+            int previousCsmElementIndex = csmElementListIterator.previousIndex();
+            CsmElement csmElement = csmElementListIterator.next();
+            int nextCsmElementIndex = csmElementListIterator.nextIndex();
+
+            Map<MatchClassification, Integer> potentialMatches = new EnumMap(MatchClassification.class);
+            for (int i = startIndex; i<nodeText.getElements().size(); i++){
+                if (!correspondingIndices.contains(i)) {
+                    TextElement textElement = nodeText.getTextElement(i);
+
+                    boolean isCorresponding = isCorrespondingElement(textElement, csmElement, node);
+
+                    if (isCorresponding) {
```

Comment:
Everything within this if statement accounts for the main change.
1. checking of the the previous and next element
2. keeping the potential matches